allennlp/data/iterators/lm_iterator.py

```diff
@@ -0,0 +1,53 @@
+from typing import Iterable, Dict, Iterator, Optional, List
+import logging
+import math
+import random
+
+from overrides import overrides
+
+from allennlp.common import Params
+from allennlp.common.util import ensure_list, is_lazy, lazy_groups_of
+from allennlp.data.instance import Instance
+from allennlp.data.iterators.data_iterator import DataIterator
+from allennlp.data.dataset import Batch
+
+# from ipdb import set_trace
+
+@DataIterator.register("lm")
+class LMIterator(DataIterator):
```

Comment:
there's nothing about this iterator that's intrinsic to language modeling? Is there a more generic name we could call it, maybe `StridedIterator` or something like that that indicates what it does?